src/tss2-tcti/tctildr.c

```diff
@@ -460,14 +460,28 @@ TSS2_RC
 Tss2_TctiLdr_Initialize (const char *nameConf,
                          TSS2_TCTI_CONTEXT **tctiContext)
 {
-    char name [PATH_MAX] = { 0, }, conf [PATH_MAX] = { 0, };
+    char *name = NULL;
+    char *conf = NULL;
     TSS2_RC rc;
+    size_t combined_length;
 
     if (nameConf == NULL) {
         return Tss2_TctiLdr_Initialize_Ex (NULL, NULL, tctiContext);
     }
+
+    combined_length = strlen (nameConf);
+    if (combined_length > PATH_MAX - 1) {
+        LOG_ERROR ("combined conf length must be between 0 and PATH_MAX");
+        return TSS2_TCTI_RC_BAD_VALUE;
+    }
+    name = calloc(combined_length + 1, sizeof(char));
+    conf = calloc(combined_length + 1, sizeof(char));
```

Comment:
NULL checks right? AFAICT `tctildr_conf_parse` will just not copy the parsed data out of nameConf if name or conf is NULL.

```C
    if (name != NULL && name_conf[0] != '\0' && name_conf[0] != ':') {
        /* name is more than empty string */
        size_t name_length = split - name_conf;
        if (name_length > PATH_MAX) {
            return TSS2_TCTI_RC_BAD_VALUE;
        }
        memcpy (name, name_conf, name_length);
        name [name_length] = '\0';
        LOG_DEBUG ("TCTI name: \"%s\"", name);
    }
    if (conf != NULL && split && split [1] != '\0') {
        /* conf is more than empty string */
        strcpy (conf, &split [1]);
        LOG_DEBUG ("TCTI conf: \"%s\"", conf);
    }

    return TSS2_RC_SUCCESS;
```

Reply:
> NULL checks right? AFAICT `tctildr_conf_parse` will just not copy the parsed data out of nameConf if name or conf is NULL.
> 
> ```c
>     if (name != NULL && name_conf[0] != '\0' && name_conf[0] != ':') {
>         /* name is more than empty string */
>         size_t name_length = split - name_conf;
>         if (name_length > PATH_MAX) {
>             return TSS2_TCTI_RC_BAD_VALUE;
>         }
>         memcpy (name, name_conf, name_length);
>         name [name_length] = '\0';
>         LOG_DEBUG ("TCTI name: \"%s\"", name);
>     }
>     if (conf != NULL && split && split [1] != '\0') {
>         /* conf is more than empty string */
>         strcpy (conf, &split [1]);
>         LOG_DEBUG ("TCTI conf: \"%s\"", conf);
>     }
> 
>     return TSS2_RC_SUCCESS;
> ```
Thanks for your review. If calloc failed, it will return error.